src/ext/java/org/opentripplanner/ext/siri/SiriTimetableSnapshotSource.java

```diff
@@ -71,7 +71,7 @@ public SiriTimetableSnapshotSource(
         parameters,
         () -> LocalDate.now(transitModel.getTimeZone())
       );
-    this.transitService = new DefaultTransitService(transitModel);
+    this.transitService = new DefaultTransitService(transitModel, getTimetableSnapshotBuffer());
```

Comment:
This feels a bit strange to me, to hand the mutable snapshot buffer to the TransitService, apparently for use in preparing API responses. I think this is related to the question you raised in your recent comment, which we'll discuss today or Thursday.

Reply:
This is indeed a bit confusing. There are 2 use cases for the transit service:
* short-lived, request-scoped instances that point to a frozen snapshot, they are used in API calls.
* long-lived instances that are used in an updater and point to the snapshot buffer itself for "transactional integrity" in the context of that updater.

I renamed the field to transitEditorService for clarity and added Javadoc.